samcli/commands/pipeline/init/interactive_init_flow.py

```diff
@@ -0,0 +1,250 @@
+from samcli.lib.cookiecutter.template import Template
+from samcli.lib.utils import osutils
+from samcli.lib.utils.git_repo import GitRepo, CloneRepoException
+from .pipeline_templates_manifest import Provider, PipelineTemplateMetadata, PipelineTemplatesManifest
+
+LOG = logging.getLogger(__name__)
+shared_path: Path = global_cfg.config_dir
+APP_PIPELINE_TEMPLATES_REPO_URL = "https://github.com/aws/aws-sam-cli-pipeline-init-templates.git"
+APP_PIPELINE_TEMPLATES_REPO_LOCAL_NAME = "aws-sam-cli-app-pipeline-templates"
+CUSTOM_PIPELINE_TEMPLATE_REPO_LOCAL_NAME = "custom-pipeline-template"
+SAM_PIPELINE_TEMPLATE_SOURCE = "AWS Quick Start Pipeline Templates"
+CUSTOM_PIPELINE_TEMPLATE_SOURCE = "Custom Pipeline Template Location"
+
+
+def do_interactive() -> None:
+    """
+    An interactive flow that prompts the user for pipeline template (cookiecutter template) location, downloads it,
+    runs its specific questionnaire then generates the pipeline config file based on the template and user's responses
+    """
+    pipeline_template_source_question = Choice(
+        key="pipeline-template-source",
+        text="Which pipeline template source would you like to use?",
+        options=[SAM_PIPELINE_TEMPLATE_SOURCE, CUSTOM_PIPELINE_TEMPLATE_SOURCE],
+    )
+    source = pipeline_template_source_question.ask()
+    if source == CUSTOM_PIPELINE_TEMPLATE_SOURCE:
+        _generate_from_custom_location()
+    else:
+        _generate_from_app_pipeline_templates()
+    click.echo("Successfully created the pipeline configuration file(s)")
+
+
+def _generate_from_app_pipeline_templates() -> None:
+    """
+    Prompts the user to choose a pipeline template from SAM predefined set of pipeline templates hosted in the git
+    repository: aws/aws-sam-cli-pipeline-init-templates.git
+    downloads locally, then generates the pipeline config file from the selected pipeline template.
+    """
+    pipeline_templates_local_dir: Path = _clone_app_pipeline_templates()
+    pipeline_templates_manifest: PipelineTemplatesManifest = _read_app_pipeline_templates_manifest(
+        pipeline_templates_local_dir
+    )
+    # The manifest contains multiple pipeline-templates so select one
+    selected_pipeline_template_metadata: PipelineTemplateMetadata = _prompt_pipeline_template(
+        pipeline_templates_manifest
+    )
+    selected_pipeline_template_dir: Path = pipeline_templates_local_dir.joinpath(
+        selected_pipeline_template_metadata.location
+    )
+    _generate_from_pipeline_template(selected_pipeline_template_dir)
+
+
+def _generate_from_custom_location() -> None:
+    """
+    Prompts the user for a custom pipeline template location, downloads locally, then generates the pipeline config file
+    """
+    pipeline_template_git_location: str = click.prompt("Template Git location")
+    if os.path.exists(pipeline_template_git_location):
+        _generate_from_pipeline_template(Path(pipeline_template_git_location))
```

Comment:
A comment here can help understand why we check a git location exists or not, is to check whether it is a local path right?